DependencyInjection/Configuration.php

```diff
@@ -185,136 +187,230 @@ private function configureClients(ArrayNodeDefinition $root)
-                                ->isRequired()
-                                ->cannotBeEmpty()
-                            ->end()
+                    ->info('Reference to a plugin service')
+                    ->children()
+                        ->scalarNode('id')
+                            ->info('Service id of a plugin')
+                            ->isRequired()
+                            ->cannotBeEmpty()
                         ->end()
-                    ->end() // End history plugin
-
-                    ->arrayNode('logger')
-                    ->canBeDisabled()
+                    ->end()
+                ->end()
+                ->arrayNode('add_host')
+                    ->canBeEnabled()
                     ->addDefaultsIfNotSet()
-                        ->children()
-                            ->scalarNode('logger')
-                                ->info('This must be a service id to a service implementing Psr\Log\LoggerInterface')
-                                ->defaultValue('logger')
-                                ->cannotBeEmpty()
-                            ->end()
-                            ->scalarNode('formatter')
-                                ->info('This must be a service id to a service implementing Http\Message\Formatter')
-                                ->defaultNull()
-                            ->end()
+                    ->info('Configure the AddHostPlugin for this client.')
+                    ->children()
+                        ->scalarNode('host')
+                            ->info('Host name including protocol and optionally the port number, e.g. https://api.local:8000')
+                            ->isRequired()
+                            ->cannotBeEmpty()
                         ->end()
-                    ->end() // End logger plugin
-
-                    ->arrayNode('redirect')
-                    ->canBeDisabled()
-                    ->addDefaultsIfNotSet()
-                        ->children()
-                            ->scalarNode('preserve_header')->defaultTrue()->end()
-                            ->scalarNode('use_default_for_multiple')->defaultTrue()->end()
+                        ->scalarNode('replace')
+                            ->info('Whether to replace the host if request already specifies it')
+                            ->defaultValue(false)
                         ->end()
-                    ->end() // End redirect plugin
+                    ->end()
+                ->end()
 
-                    ->arrayNode('retry')
-                    ->canBeDisabled()
-                    ->addDefaultsIfNotSet()
-                        ->children()
-                            ->scalarNode('retry')->defaultValue(1)->end()
-                        ->end()
-                    ->end() // End retry plugin
+                // TODO add aditional plugins that are only usable on a specific client
```

Comment:
ContentLength is globally available and has no configuration options. did you mean something else?